src/waveform/waveform.h

```diff
@@ -31,6 +31,13 @@ union WaveformData {
 
 class Waveform {
   public:
+    enum class SaveState {
+        Virgin = 0,
```

Comment:
I'm not a native speaker, but "Virgin" sounds very strange to me. And it doesn't reveal what state the waveform actually has. Please use the term "Empty", because that's how it is actually used when looking at the code.